tests/run.rs

```diff
@@ -800,3 +800,122 @@ fn complains_when_expected_an_unmocked_command() -> R<()> {
     )?;
     Ok(())
 }
+
+mod file_mocking {
+    use super::*;
+
+    mod in_ruby {
+        use super::*;
+
+        #[test]
+        fn allows_to_mock_files_existence_in_ruby() -> R<()> {
+            test_run(
+                r##"
+                    |#!/usr/bin/env ruby
+                    |`cat /foo/bar` if File.exists?("/foo/bar")
+                "##,
+                r##"
+                    |protocols:
+                    |  - protocol:
+                    |      - /bin/cat /foo/bar
+                    |    filetree:
+                    |      - /foo/bar
+                "##,
+                Ok(()),
+            )?;
+            Ok(())
+        }
+
+        #[test]
+        fn allows_to_mock_folders_existence_in_ruby() -> R<()> {
+            test_run(
+                r##"
+                    |#!/usr/bin/env ruby
+                    |`cat /foo/bar/test` if Dir.exists?("/foo/bar/")
+                    |`cat /foo/bar/test2` if File.directory?("/foo/bar/")
+                "##,
+                r##"
+                    |protocols:
+                    |  - protocol:
+                    |      - /bin/cat /foo/bar/test
+                    |      - /bin/cat /foo/bar/test2
+                    |    filetree:
+                    |      - /foo/bar/
+                "##,
+                Ok(()),
+            )?;
+            Ok(())
+        }
+    }
```

Comment:
If ruby and bash access the file system in the same way, I wouldn't add any ruby tests for this. (I only added one ruby test for spawning subprocesses, since ruby uses different syscalls for that than bash.)

Reply:
in other words, it's safe to assume that if ruby uses `stat` and `statbuf` to check for file existence, they would do the same for directories?

Reply:
Sorry, I wasn't just talking about the directory tests, but about all ruby tests that you added. So far I only added non-bash tests when other languages behaved differently from bash. In this case it seems that both bash and ruby access the file system in the same way (both for files and directories) so I would consider just sticking to bash in the tests.

Reply:
got it, that answers my other question then about `fail if...`